MultiLabel/Core/src/main/java/org/tribuo/multilabel/evaluation/MultiLabelConfusionMatrix.java

```diff
@@ -158,15 +159,18 @@ public double confusion(MultiLabel predicted, MultiLabel truth) {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder();
-        sb.append("[");
-        for (int i = 0; i < mcm.length; i++) {
-            DenseMatrix cm = mcm[i];
-            sb.append(cm.toString());
-            sb.append("\n");
-        }
-        sb.append("]");
-        return sb.toString();
+        return getDomain().getDomain().stream()
```

Comment:
This is sensitive to the iteration order of the HashSet inside the domain, whereas the old one was fixed on construction (though when it was constructed it was fixed based on the iteration order of the HashSet in the domain).

This comment is more a todo for later when the label domains have consistent iteration order (probably lexicographic like the features).

Reply:
Ah, good point. That variable iteration order has confused me several times. There's always `LinkedHashSet`? Or maybe https://docs.oracle.com/javase/9/docs/api/java/util/Set.html#of-E...- would have stable iteration order like Guava's `ImmutableHashSet`? How long will Java 8 be required?